docs/modules/secret-operator/pages/secretclass.adoc

```diff
@@ -38,6 +38,21 @@ CAUTION: Attributes of the certificate (such as the expiration date, fingerprint
 
 xref:scope.adoc[Scopes] are used to populate the claims (such as `subjectAlternateName`) of the provisioned certificates.
 
+[#backend-tls-key-length]
+=== `TLS key length`
+
+Currently, only RSA is supported in the `autoTls` backend. You can however configure the key length for generated private keys or certs.
+
+----
+include::example$secretclass-tls-key-length.yaml[]
+----
+<1> `autoTls.ca.keyGeneration` specifies which algorithm and additional parameters are used
+<2> `autoTls.ca.keyGeneration.rsa` specifies the RSA algorithm (RSA currently is the only one supported)
+<3> `autoTls.ca.keyGeneration.rsa.length` specifies the amount of bits used for key or certs. Currently `2048`, `4096` and `8192` are supported.
+
+CAUTION: Using a key length higher than `2048` will significantly increase the computation time. If options higher than `2048` are chosen, the CPU 
+         resources for the secret operator should be increased in order to avoid pods waiting for the computation of their key.
```

Comment:
```suggestion
CAUTION: Using a key length higher than `2048` will significantly increase the computation time to create new key-pairs.
The SSL Labs https://github.com/ssllabs/research/wiki/SSL-and-TLS-Deployment-Best-Practices[SSL and TLS Deployment Best Practices] as of 2024-10-01 recommend

>  For most web sites, using RSA keys stronger than 2,048 bits and ECDSA keys stronger than 256 bits is a waste of CPU power and might impair user experience

If options higher than `2048` are chosen, the CPU resources for the secret operator should be increased in order to avoid Pods being stuck in `Pending` waiting for the computation of their key-pair.
```

Reply:
Yeah i like that thanks